release_process.md

```diff
@@ -1,162 +1,550 @@
-# Release process
+# Release Process
+
+Lance uses a semantic versioning release workflow with Release Candidates (RC) for voting before stable releases. The process features linear commit history on branches.
+
+## Overview
+
+The release process follows these stages:
+1. **Development** on `main` branch with beta versions
+2. **Preview Releases** published from beta versions (beta.1+)
+3. **Release Candidates** created for voting (rc.1, rc.2, etc.)
+4. **Stable Releases** promoted from approved RCs
+5. **Patch Releases** managed through release branches
+
+## Version Semantics
+
+### Version Format
+
+Lance uses semantic versioning with prerelease identifiers:
+- **Stable**: `X.Y.Z` (e.g., `1.3.0`)
+- **Beta**: `X.Y.Z-beta.N` (e.g., `1.3.0-beta.1`, `1.3.0-beta.2`)
+- **RC**: `X.Y.Z-rc.N` (e.g., `1.3.0-rc.1`, `1.3.0-rc.2`)
+
+### Beta Version States
+
+- **beta.0**: Unreleased version (exists on branch but not published)
+  - Created after cutting an RC to mark the next unreleased version
+  - Indicates no preview has been published yet
+- **beta.1+**: Published preview releases
+  - Created when publishing beta preview artifacts
+
+### Publishing Channels
+
+**Stable Releases (no suffix)**:
+- **Rust**: crates.io
+- **Python**: PyPI
+- **Java**: Maven Central
+- **Protobuf**: Buf Schema Registry
+
+**RC Releases (`-rc.N`)**:
+- **Rust**: Not published (use git tag)
+- **Python**: fury.io
+- **Java**: Maven Central (pre-release)
+- **Protobuf**: Buf Schema Registry (pre-release)
+
+**Beta Releases (`-beta.N`)**:
+- **Rust**: Not published (use git tag)
+- **Python**: fury.io
+- **Java**: Maven Central (pre-release)
+- **Protobuf**: Buf Schema Registry (pre-release)
+
+### GitHub Releases and Release Notes
+
+**RC Releases (`-rc.N`)**:
+- Created as GitHub Pre-Releases with custom-generated release notes
```

Comment:
I find "custom-generated" release notes unclear. Does that mean a person wrote them? Or a script?

I think the goal of this section should be to make clear how and when release notes are created. If there is a script that generates them, we should link to that.